.github/workflows/test.yaml

```diff
@@ -60,32 +59,36 @@ jobs:
       - name: Test Setup
         uses: ./.github/actions/test-setup
         with:
-          should-install-bun: true
-
-      - name: Publish PR to NPM
-        if: ${{ github.event.pull_request.head.repo.full_name == github.repository }}
-        id: release
-        uses: ./.github/actions/pr-release
-        with:
-          npm-token: ${{ secrets.NPM_TOKEN }}
-          github-token: ${{ secrets.GITHUB_TOKEN }}
+          should-install-bun: github.ref_name == 'master' || github.head_ref == 'changesets-release/master'
 
-      - name: Echo Published Version
-        run: echo ${{ steps.release.outputs.published_version }}
+      - name: Run UI tests (on template)
+        run: sh ./scripts/tests-ui.sh
 
-      - name: Pretest
-        run: pnpm pretest
+      - name: Wait for release to succeed
+        if: github.ref_name == 'master' || github.head_ref == 'changesets-release/master'
+        uses: lewagon/wait-on-check-action@v1.3.4
+        with:
+          ref: ${{ github.ref }}
```

Comment:
From the [environment variable docs](https://docs.github.com/en/actions/learn-github-actions/variables) for `GITHUB_REF`:
> 	The fully-formed ref of the branch or tag that triggered the workflow run. For workflows triggered by push, this is the branch or tag ref that was pushed. For workflows triggered by pull_request, this is the pull request merge branch. For workflows triggered by release, this is the release tag created. For other triggers, this is the branch or tag ref that triggered the workflow run. This is only set if a branch or tag is available for the event type. The ref given is fully-formed, meaning that for branches the format is refs/heads/<branch_name>, for pull requests it is refs/pull/<pr_number>/merge, and for tags it is refs/tags/<tag_name>. For example, refs/heads/feature-branch-1.

I think that this won't work as expected for changeset PRs because the step will be waiting for the `Release` workflow on `github.ref` which would be `refs/pull/<pr_number>/merge` and there won't be such a workflow running because `Release` is running only on `push` to `master` and `release/*` branches.

 I think that the value here should be hardcoded to `refs/heads/master`.